libfwupd/fwupd-bios-attr.c

```diff
@@ -0,0 +1,878 @@
+	guint64 scalar_increment;
+	GPtrArray *possible_values;
+} FwupdBiosAttrPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE(FwupdBiosAttr, fwupd_bios_attr, G_TYPE_OBJECT)
+#define GET_PRIVATE(o) (fwupd_bios_attr_get_instance_private(o))
+
+/**
+ * fwupd_bios_attr_get_lower_bound:
+ * @self: a #FwupdBiosAttr
+ *
+ * Gets the lower bound for integer attributes or minimum length for string attributes
+ *
+ * Returns: guint64
+ *
+ * Since: 1.8.4
+ **/
+guint64
+fwupd_bios_attr_get_lower_bound(FwupdBiosAttr *self)
+{
+	FwupdBiosAttrPrivate *priv = GET_PRIVATE(self);
+	g_return_val_if_fail(FWUPD_IS_BIOS_ATTR(self), 0);
+	return priv->lower_bound;
+}
+
+/**
+ * fwupd_bios_attr_get_upper_bound:
+ * @self: a #FwupdBiosAttr
+ *
+ * Gets the upper bound for integer attributes or maximum length for string attributes
+ *
+ * Returns: guint64
+ *
+ * Since: 1.8.4
+ **/
+guint64
+fwupd_bios_attr_get_upper_bound(FwupdBiosAttr *self)
+{
+	FwupdBiosAttrPrivate *priv = GET_PRIVATE(self);
+	g_return_val_if_fail(FWUPD_IS_BIOS_ATTR(self), 0);
+	return priv->upper_bound;
+}
+
+/**
+ * fwupd_bios_attr_get_scalar_increment:
+ * @self: a #FwupdBiosAttr
+ *
+ * Gets the scalar increment used for integer attributes
+ *
+ * Returns: guint64
+ *
+ * Since: 1.8.4
+ **/
+guint64
+fwupd_bios_attr_get_scalar_increment(FwupdBiosAttr *self)
+{
+	FwupdBiosAttrPrivate *priv = GET_PRIVATE(self);
+	g_return_val_if_fail(FWUPD_IS_BIOS_ATTR(self), 0);
+	g_return_val_if_fail(priv->type != FWUPD_BIOS_ATTR_TYPE_INTEGER, 0);
+	return priv->scalar_increment;
+}
+
+/**
+ * fwupd_bios_attr_set_upper_bound:
+ * @self: a #FwupdBiosAttr
+ * @val: a guint64 value to set bound to
+ *
+ * Sets the upper bound used for BIOS integer attributes or max length for string attributes
+ *
+ * Since: 1.8.4
+ **/
+void
+fwupd_bios_attr_set_upper_bound(FwupdBiosAttr *self, guint64 val)
+{
+	FwupdBiosAttrPrivate *priv = GET_PRIVATE(self);
+	g_return_if_fail(FWUPD_IS_BIOS_ATTR(self));
+	priv->upper_bound = val;
+}
+
+/**
+ * fwupd_bios_attr_set_lower_bound:
+ * @self: a #FwupdBiosAttr
+ * @val: a guint64 value to set bound to
+ *
+ * Sets the lower bound used for BIOS integer attributes or max length for string attributes
+ *
+ * Since: 1.8.4
+ **/
+void
+fwupd_bios_attr_set_lower_bound(FwupdBiosAttr *self, guint64 val)
+{
+	FwupdBiosAttrPrivate *priv = GET_PRIVATE(self);
+	g_return_if_fail(FWUPD_IS_BIOS_ATTR(self));
+	priv->lower_bound = val;
+}
+
+/**
+ * fwupd_bios_attr_set_scalar_increment:
+ * @self: a #FwupdBiosAttr
+ * @val: a guint64 value to set increment to
+ *
+ * Sets the scalar increment used for BIOS integer attributes
+ *
+ * Since: 1.8.4
+ **/
+void
+fwupd_bios_attr_set_scalar_increment(FwupdBiosAttr *self, guint64 val)
+{
+	FwupdBiosAttrPrivate *priv = GET_PRIVATE(self);
+	g_return_if_fail(FWUPD_IS_BIOS_ATTR(self));
+	priv->scalar_increment = val;
+}
+
+/**
+ * fwupd_bios_attr_get_kernel_type:
+ * @self: a #FwupdBiosAttr
+ *
+ * Gets the BIOS attribute type used by the kernel interface
```

Comment:
needs a trailing `.` -- there are a few more in this file too.

Reply:
fixed